sources/api/thar-be-settings/src/config.rs

```diff
@@ -3,11 +3,14 @@ use crate::{error, Result};
 use itertools::join;
 use snafu::{ensure, ResultExt};
 use std::collections::HashSet;
-use std::fs;
+use std::fs::{self, OpenOptions};
+use std::io::prelude::*;
+use std::os::unix::fs::OpenOptionsExt;
 use std::path::{Path, PathBuf};
 use std::process::Command;
 
 const SYSTEMCTL_DAEMON_RELOAD: &str = "systemctl daemon-reload";
+const DEFAULT_FILE_MODE: u32 = 0o644;
```

Comment:
This will be readable by all groups? I'm curious the reasoning behind this permission mode as the default? My gut instinct would be to have a `0600` (which it looks like you've set elsewhere) to prevent other users from reading. Granted, this is weird and nuanced since Bottlerocket isn't a multi-user system anyways but curious to hear your thoughts.

Reply:
That is the existing default mode, so I didn't want to change any behavior by adding the ability to set it to something other than that. I agree it ultimately doesn't make much difference, but wanted to minimize any possible side effects.